python/lsst/ts/observing/utilities/logging.py

```diff
@@ -0,0 +1,143 @@
+}
+
+
+def get_logger(logger_name=None, use_color=True):
+    """
+    Return a logger with the "logger_name".
+
+    Parameters
+    ----------
+    logger_name : str, optional
+        The logger name to be used in different contexts.
+    use_color : bool, optional
+        Use colors on Stream Loggers.
+
+    Returns
+    ------
+    logging.Logger : the logger to be used.
+    """
+    message_format = " [%(levelname).1s %(asctime)s %(name)s] %(message)s"
+    date_format = "%Y-%m-%d %H:%M:%S"
+
+    _logger = logging.getLogger(logger_name)
+
+    if len(_logger.handlers) == 0:
+        formatter = MyLogFormatter(message_format, datefmt=date_format, use_colors=use_color)
+
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setFormatter(formatter)
+
+        _logger.addHandler(handler)
+        _logger.setLevel(logging.DEBUG)
+
+    return _logger
+
+
+class MyLogFormatter(logging.Formatter):
+    """
+    Custom log formatter inspired on Jupyter's logging messages.
+
+    Parameters
+    ----------
+    fmt : str, optional
+        Log format.
+    use_colors : bool (True).
+        Print colored messages?
+    """
+    def __init__(self, fmt=" [%(levelname).1s %(asctime)s %(name)s] %(message)s", datefmt="%Y-%m-%d %H:%M:%S",
+                 use_colors=True):
+        logging.Formatter.__init__(self, fmt, datefmt=datefmt)
+        self.use_colors = use_colors
+
+    @staticmethod
+    def color_format(message, levelname, left_char="[", right_char="]"):
+        """
+        Replaces part of the output message with characters that start and end
+        the colored string.
+
+        Parameters
+        ----------
+        message : str
+            Logging message.
+        levelname : str
+            Logging level name as a string.
+        left_char : str, optional
+            Left character that encapsulates the left side of the part that
+            will be colored. Default: "[".
+        right_char : str, optional
+            Right character that encapsulates the right side of the part that
+            will be colored. Default: "[".
+
+        Returns
+        -------
+        str : message containing the special characters that enable colors.
+        """
+        colour = COLOR_SEQ % (30 + COLORS[levelname])
+
+        message = message.replace(left_char, "{:s} {:s}".format(colour, left_char))
+        message = message.replace(right_char, "{:s} {:s}".format(right_char, RESET_SEQ))
+
+        return message
+
+    def format(self, record):
+        """
+        Override the default logging format.
+
+        See also
+        --------
+        https://docs.python.org/3/library/logging.html#logging.Formatter.format
+        """
+        # Call the original formatter class to do the grunt work
+        result = logging.Formatter.format(self, record)
+
+        if self.use_colors:
+            result = self.color_format(result, record.levelname)
+
+        return result
+
+
+if __name__ == "__main__":
```

Comment:
I think you can remove this part here. We usually don't leave these in library modules.